mypy/expandtype.py

```diff
@@ -96,8 +96,12 @@ def visit_type_var(self, t: TypeVarType) -> Type:
             return repl
 
     def visit_callable_type(self, t: CallableType) -> Type:
+        extra = {}  # type: Dict[str, Any]
+        if t.type_guard is not None:
+            extra['type_guard'] = t.type_guard.accept(self)
         return t.copy_modified(arg_types=self.expand_types(t.arg_types),
-                               ret_type=t.ret_type.accept(self))
+                               ret_type=t.ret_type.accept(self),
+                               **extra)
```

Comment:
Using `**extra` could be slow here, since it's unoptimized in mypyc. Why not pass `None` here if no type guard is defined, avoiding the use of `**extra` and the creation of a dictionary?